envsubst.go

```diff
@@ -10,29 +10,30 @@ import (
 // String returns the parsed template string after processing it.
 // If the parser encounters invalid input, it returns an error describing the failure.
 func String(s string) (string, error) {
-	return StringRestricted(s, false, false)
+	return StringRestricted(s, false, false, false)
 }
 
 // StringRestricted returns the parsed template string after processing it.
 // If the parser encounters invalid input, or a restriction is violated, it returns
 // an error describing the failure.
-func StringRestricted(s string, noUnset, noEmpty bool) (string, error) {
+// Errors on first failure or returns a collection of failures if failOnFirst is false
+func StringRestricted(s string, noUnset, noEmpty, failOnfirst bool) (string, error) {
```

Comment:
Let's remove these API changes, and keep only the parser code (and the cli that invokes the parser of course).